api/v1/gitrepository_types.go

```diff
@@ -38,6 +38,21 @@ const (
 	IncludeUnavailableCondition string = "IncludeUnavailable"
 )
 
+// GitVerificationMode specifies the verification mode for a Git repository.
+type GitVerificationMode string
```

Comment:
```suggestion
type GitVerificationMode string

func (m GitVerificationMode) Valid() bool {
	switch m {
	case ModeHead, ModeTag, ModeTagAndHead:
		return true
	default:
		return false
	}
}
```

(or an `error` return).